2-task/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+/*
+To test PUT Method
+curl -X PUT \
+-H "Content-Type: application/json" \
+-d  '{"name":"Rage", "author": "Stephen King", "year": "1000"}' \
+localhost:10000/books/put
+*/
+
+import (
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sort"
+	"strconv"
+)
+
+func main() {
+	books := &BookList{
+		Books: []Book{
+			Book{Name: "Rage", Author: "Stephen King", Year: "1977"},
+			Book{Name: "Philosopher's Stone", Author: "J. K. Rowling", Year: "1997"},
+			Book{Name: "All Quiet on the Western Front", Author: "Erich Maria Remarque", Year: "1929"},
+		},
+	}
+	sort.Sort(sortedBooks(books.Books))
+	handleRequests(books)
+}
+
+func handleRequests(books BookStore) {
+	myRouter := mux.NewRouter().StrictSlash(true)
+	myRouter.HandleFunc("/books", func(w http.ResponseWriter, r *http.Request) {
+		returnAllBooks(w, r, books)
+	})
```

Comment:
I know that by default it is set to get but try to provide it again - it is more representative
```suggestion
	}).Methods(http.MethodGet)
```